### PR TITLE
[WIP] initial draft of Fine Grained Permissions JEP

### PR DESCRIPTION
**Pull request by mikecirioli**

### CODE REVIEW

File: jep/401/README.adoc [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+[TIP]
+====
+Explain why the existing code base or process is inadequate to address the problem that the JEP solves.
+This section may also contain any historical context such as how things were done before this proposal.
+
+* Do not discuss design choices or alternative designs that were rejected - those belong in the Reasoning section.
+====
+
+The current permission model provides a false sense of security for Jenkins administrators.  The motivation was to allow some users the ability to configure limited aspects of a Jenkins instance without granting them full `Jenkins.ADMINISTRATOR` privelleges, however it is trivial for a motivated user to grant themselves the additional priveleges by means of the groovy console (`Jenkins.RUN_SCRIPTS`), or by means of a malicious plugin (`Jenkins.UPLOAD_PLUGINS`, `Jenkins.CONFIGURE_UPDATECENTER`).
```
Comment (aHenryJard) [inline jep/401/README.adoc]: I think we can use

File: jep/401/README.adoc [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+The following permission types are now marked as `@Deprecated` and `@Restricted(NoExternalUse.class)`:
+
+* `Jenkins.RUN_SCRIPTS`
+* `Jenkins.CONFIGURE_UPDATECENTER`
+* `Jenkins.UPLOAD_PLUGINS`
+
+This allows for backwards compatiblilty with plugins while providing motivation to refactor any usages of the deprecated checks in future releases.
+
+Actions that formerlly checked for these permissions will instead check for `Jenkins.ADMINISTER`.
```
Comment (HCloward) [inline jep/401/README.adoc]: looks like you got an extra "l" in formerly

File: jep/401/README.adoc [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+[TIP]
+====
+Explain why the existing code base or process is inadequate to address the problem that the JEP solves.
+This section may also contain any historical context such as how things were done before this proposal.
+
+* Do not discuss design choices or alternative designs that were rejected - those belong in the Reasoning section.
+====
+
+The current permission model provides a false sense of security for Jenkins administrators.  The motivation was to allow some users the ability to configure limited aspects of a Jenkins instance without granting them full `Jenkins.ADMINISTRATOR` privelleges, however it is trivial for a motivated user to grant themselves the additional priveleges by means of the groovy console (`Jenkins.RUN_SCRIPTS`), or by means of a malicious plugin (`Jenkins.UPLOAD_PLUGINS`, `Jenkins.CONFIGURE_UPDATECENTER`).
```
Comment (HCloward) [inline jep/401/README.adoc]: "privelleges" == privileges
Before "however" replace the comma with a semicolon to fix a comma splice - I would also add a comma after "however"
"priveleges" == privileges

File: jep/401/README.adoc [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+By consolidating all permission types that effectively allow a user to have full `root` access on a Jenkins instance, and introducing a new permission (`Jenkins.CONFIGURE`) that allows a limited amount of access to configure certain non-critical functionality, a Jenkins administrator can safely delegate configuration aspects while being confident that security concerns are being met.
+
+The `Jenkins.CONFIGURE` permission type is not intended to replace any `Item` level permissions.  It is soley for the purpose of sensistive, security related Jenkins configuration settings from those that a non-root administrator might be expected to manage.
```
Comment (HCloward) [inline jep/401/README.adoc]: "soley" == solely
"It is solely for the purpose of . . . " separating, removing? . . . "sensitive, security-related Jenkins configuration settings . . . "

File: jep/401/README.adoc [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+Explain why particular design decisions were made.
+Describe alternate designs that were considered and related work. For example, how the feature is supported in other systems.
+Provide evidence of consensus within the community and discuss important objections or concerns raised during discussion.
+
+* Use sub-headings to organize this section for ease of readability.
+* Do not talk about history or why this needs to be done - that is part of Motivation section.
+====
+
+=== Consolidating "dangerous" permissions
+The permissions that are considered "dangerous" effectively provide the user with the means to grant themselves the "ultimate" permission, `Jenkins.ADMINISTER`, therby making their existence redundant.  Consolidating these permissions into `Jenkins.ADMINISTER` makes it unambiguous as to what a user is or is not permitted to do.
```
Comment (HCloward) [inline jep/401/README.adoc]: "therby" == thereby
"Consolidating these permissions into `Jenkins.ADMINISTRATOR` removes ambiguity concerning what a user is or is not permitted to do."

**Review by HCloward — COMMENTED** [review]
Found a couple of typos, but overall makes sense to me so far.

File: jep/401/README.adoc [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+====
+Give a short (200 word) description of the technical issue addressed.
+
+* Use present tense - describe what the proposal "does" (as if it were already done), not what it will do.
+* Do not go into technical details and instead put those in the Specification section.
+* Do not talk about history or why this needs to be done. Instead, add the history to the Motivation section.
+====
+The permission types `Jenkins.RUN_SCRIPTS`, `Jenkins.CONFIGURE_UPDATECENTER`, and `Jenkins.UPLOAD_PLUGINS` provide a false sense of security, as a user is able to escalate this limited privilege to `Jenkins.ADMINISTER`.  These permissions are being marked as `@Deprecated` and are now consolidated as `Jenkins.Administer`.  
+
+A new permission type, `Jenkins.CONFIGURE` (implied by `Jenkins.ADMINISTER`) has been created for actions that allow for the management of Jenkins configuration items that do not require `Jenkins.ADMINISTER`.
```
Comment (aHenryJard) [inline jep/401/README.adoc]: you have to use present -> is created

Comment (aHenryJard) [reply jep/401/README.adoc]: I wonder if we should add something on the fact that now, ADMINISTER is keep to sensible operations (vs before administer was for all administer operations). To be clear.

**Review by aHenryJard — COMMENTED** [review]
Did you find any JIRA interesting to be referenced here at the end of the JEP ?

**Pull request merged**